pkg/util/tracing/tracer.go

```diff
@@ -646,14 +646,46 @@ func (t *Tracer) ExtractMetaFrom(carrier Carrier) (*SpanMeta, error) {
 	}, nil
 }
 
-// GetActiveSpanFromID retrieves any active span given its span ID.
-func (t *Tracer) GetActiveSpanFromID(spanID uint64) (*Span, bool) {
+// GetActiveRootSpanFromID retrieves any active span given its span ID.
+func (t *Tracer) GetActiveRootSpanFromID(spanID uint64) (*Span, bool) {
 	t.activeSpans.Lock()
 	span, found := t.activeSpans.m[spanID]
 	t.activeSpans.Unlock()
 	return span, found
 }
 
+// SetAllDescendantsSpanVerbosity takes a root Span and sets all its
+// descendant spans' verbosity using setAllChildrenSpanVerbosity. The method
+// returns true if setAllChildrenSpanVerbosity is called successfully and false
+// if not because the root Span was a no-op span.
+func (t *Tracer) SetAllDescendantsSpanVerbosity(rootSpan *Span, to bool) bool {
```

Comment:
Ok, I've set the root span verbosity inside the new method, `spanInner.SetVerboseRecursively`, which is called from `Span.SetVerboseRecursively`. It does so by calling into `spanInner.SetVerbose`, which already panic and errors out on a no-op span, so this matches with your expectation that the caller should guarantee the span isn't a no-op. 

I had to move the meat of the locking/unlocking/recursion to `crdbSpan` instead, because we actually can't recurse on `span` or `spanInner` because we don't have a hold on either of those objects for the children.